lib/models/os.coffee

```diff
@@ -39,24 +239,33 @@ getOsModel = (deps, opts) ->
 	# 	console.log('The raspberry-pi image was last modified in ' + date);
 	# });
 	#
+	# resin.models.os.getLastModified('raspberrypi3', '2.0.0').then(function(date) {
+	# 	console.log('The raspberry-pi image was last modified in ' + date);
+	# });
+	#
 	# resin.models.os.getLastModified('raspberry-pi', function(error, date) {
 	# 	if (error) throw error;
 	# 	console.log('The raspberry-pi image was last modified in ' + date);
 	# });
 	###
-	exports.getLastModified = (deviceType, callback) ->
-		request.send
-			method: 'HEAD'
-			url: "/api/v1/image/#{deviceType}/"
-			baseUrl: imageMakerUrl
-		.catch
-			code: 'ResinRequestError'
-			statusCode: 404
-		, ->
-			throw new errors.ResinRequestError('No such device type')
-		.then (response) ->
-			return new Date(response.headers['last-modified'])
-		.asCallback(callback)
+	exports.getLastModified = (deviceType, version = 'latest', callback) ->
+		callback = findCallback(arguments)
+
+		return isValidDeviceType(deviceType)
+			.then (isValid) ->
+				if not isValid
+					throw new errors.ResinRequestError('No such device type')
+			.then ->
+				return normalizeVersion(version)
+			.then (version) ->
+				imgMakerHelper.request
+					method: 'HEAD'
+					url: "/image/#{deviceType}/?version=#{version}"
+			.catch notFoundResponse, ->
+				throw new errors.ResinRequestError('No such version for the device type')
```

Comment:
Shouldn't really be a `ResinRequestError` either. I'd just make this `Error` for now, we could add more specific cases later, but this is forward compatible, and I don't think it matters much.